feat(.NET): Screenshots for MAUI (#9120)
getsentry/sentry-docs: docs/platforms/dotnet/guides/maui/enriching-events/screenshots/index.mdx

---
title: "Screenshots"
description: "Learn more about taking screenshots when an error occurs. Sentry pairs the screenshot with the original event, giving you additional insight into issues."
---

When a user experiences an error, an exception or a crash, Sentry provides the ability to take a screenshot and include it as an <PlatformLink to="/enriching-events/attachments/">attachment</PlatformLink>.

This feature only applies to SDKs with a user interface, such as the ones for mobile and desktop applications. In some environments like native iOS, taking a screenshot requires the UI thread and in the event of a crash, that might not be available. Another example where a screenshot might not be available is when the event happens before the screen starts to load. So inherently, this feature is a best effort solution.

> 

Author: I took the updated first part but I'd like to keep the `best effort solution` in there. It has come up from time to time and this helped answer the "_but some issues don't have screenshots_" issues.

## Enabling Screenshots

> Screenshots may contain <PlatformLink to="/data-management/sensitive-data/">PII</PlatformLink> and is an opt-in feature. You can enable it as shown below:
> 
> <PlatformContent includePath="enriching-events/attach-screenshots" />
> 
> Capturing a screenshot is a synchronous operation supported on both iOS and Android, with the requirement of using the UI thread specifically on iOS.

Reviewer: ```suggestion
Because screenshots may contain <PlatformLink to="/data-management/sensitive-data/">PII</PlatformLink>, they are an opt-in feature. You can enable screenshots as shown below:

<PlatformContent includePath="enriching-events/attach-screenshots" />

Capturing a screenshot is a synchronous operation supported on both iOS and Android. For iOS, a UI thread is required.
```

## Viewing Screenshots

> A thumbnail of the screenshot shows up at the top of the **Issue Details** page:

Reviewer: ```suggestion
If one is available, you'll see a thumbnail of the screenshot if you click on a specific issue from the  [**Issues**](https://demo.sentry.io/issues/) page:
```

> ![Screenshot Thumbnail](/platforms/dotnet/guides/xamarin/enriching-events/screenshots/screenshot-thumbnail.png)

Reviewer: Would it be possible to make the screenshot a bit higher resolution? This one is a bit fuzzy. Since we've moved to Next.js, all images need to be added to the public folder :) https://docs.sentry.io/contributing/pages/images/

Author: Yes, definitely! I'll update them then.

> Additionally, screenshots appear in the "Attachments" tab, where you can view all attachments, as well as associated events. Click the event ID to open the **Issue Details** page of that specific event.
> 
> ![Screenshots List Example](/platforms/dotnet/guides/xamarin/enriching-events/screenshots/screenshot-list-example.png)
> 
> You can see an overview of all the screenshots associated with the events in your issue by clicking "Screenshots" in the "Attachments" tab.

Reviewer: ```suggestion
Once you've clicked on the event ID of specific issue, you'll also be able to view screenshots from the "Attachments" tab where you'll be able to see attachments, as well as associated events. 

![Screenshots List Example](/platforms/dotnet/guides/xamarin/enriching-events/screenshots/screenshot-list-example.png)

```
